api/layer/object.go

```diff
@@ -71,70 +66,101 @@ type (
 	f := &findParams{
-		filters: []filter{{attr: object.AttributeFileName, val: filename}},
-		cid:     cid,
-		prefix:  "",
+		attr: [2]string{object.AttributeFileName, filename},
+		cid:  cid,
 	}
 	return n.objectSearch(ctx, f)
 }
 
 // objectSearch returns all available objects by search params.
-func (n *layer) objectSearch(ctx context.Context, p *findParams) ([]*object.ID, error) {
-	var opts object.SearchFilters
-
-	opts.AddRootFilter()
-
-	for _, filter := range p.filters {
-		opts.AddFilter(filter.attr, filter.val, object.MatchStringEqual)
+func (n *layer) objectSearch(ctx context.Context, p *findParams) ([]oid.ID, error) {
+	prm := PrmObjectSelect{
+		Container:      *p.cid,
+		ExactAttribute: p.attr,
+		FilePrefix:     p.prefix,
 	}
 
-	if p.prefix != "" {
-		opts.AddFilter(object.AttributeFileName, p.prefix, object.MatchCommonPrefix)
-	}
-	searchParams := new(client.SearchObjectParams).WithContainerID(p.cid).WithSearchFilters(opts)
+	n.prepareAuthParameters(ctx, &prm.PrmAuth)
+
+	res, err := n.neoFS.SelectObjects(ctx, prm)
 
-	ids, err := n.pool.SearchObject(ctx, searchParams, n.CallOptions(ctx)...)
-	return ids, n.transformNeofsError(ctx, err)
+	return res, n.transformNeofsError(ctx, err)
 }
 
-func newAddress(cid *cid.ID, oid *object.ID) *object.Address {
-	address := object.NewAddress()
-	address.SetContainerID(cid)
-	address.SetObjectID(oid)
-	return address
+func newAddress(cid *cid.ID, oid *oid.ID) *address.Address {
+	addr := address.NewAddress()
+	addr.SetContainerID(cid)
+	addr.SetObjectID(oid)
+	return addr
 }
 
 // objectHead returns all object's headers.
-func (n *layer) objectHead(ctx context.Context, cid *cid.ID, oid *object.ID) (*object.Object, error) {
-	ops := new(client.ObjectHeaderParams).WithAddress(newAddress(cid, oid)).WithAllFields()
-	obj, err := n.pool.GetObjectHeader(ctx, ops, n.CallOptions(ctx)...)
-	return obj, n.transformNeofsError(ctx, err)
+func (n *layer) objectHead(ctx context.Context, idCnr *cid.ID, idObj *oid.ID) (*object.Object, error) {
+	prm := PrmObjectRead{
+		Container:  *idCnr,
+		Object:     *idObj,
+		WithHeader: true,
+	}
+
+	n.prepareAuthParameters(ctx, &prm.PrmAuth)
+
+	res, err := n.neoFS.ReadObject(ctx, prm)
+	if err != nil {
+		return nil, n.transformNeofsError(ctx, err)
+	}
+
+	return res.Head, nil
 }
 
-// objectGetWithPayloadWriter and write it into provided io.Reader.
-func (n *layer) objectGetWithPayloadWriter(ctx context.Context, p *getParams) (*object.Object, error) {
-	// prepare length/offset writer
-	w := newWriter(p.Writer, p.offset, p.length)
-	ops := new(client.GetObjectParams).WithAddress(newAddress(p.cid, p.oid)).WithPayloadWriter(w)
-	obj, err := n.pool.GetObject(ctx, ops, n.CallOptions(ctx)...)
-	return obj, n.transformNeofsError(ctx, err)
+// writes payload part of the NeoFS object to the provided io.Writer.
+// Zero range corresponds to full payload (panics if only offset is set).
+func (n *layer) objectWritePayload(ctx context.Context, p getParams) error {
```

Comment:
I think this name can be a little confused. It would be nice to mention that this is `read` operation. But it's a matter of taste